### PR TITLE
Improve state module docs, replace references to state.highstate/state.sls with state.apply

### LINKED ISSUE #32037: Increase the visibility of state.apply in Salt's documentation
This function, added in 2015.5.0 as a more intuitively named alternative, is not mentioned at all in the docs. To increase adoption, it should be more visible.

### PR DESCRIPTION
**Pull request by terminalmage**
Resolves #32037.

### CODE REVIEW

**Comment by terminalmage** [comment]
Would like some critique on this from one or more of the following: @cachedout, @thatch45, @UtahDave.

File: doc/topics/tutorials/states_pt1.rst [hunk 1]
```diff
@@ -125,13 +125,16 @@ Next, let's run the state we created. Open a terminal on the master and run:
...
-expressions. When it does match an expression the modules listed for it will be
-downloaded, compiled, and executed.
+Our master is instructing all targeted minions to run :func:`state.apply
+<salt.modules.state.apply>`. When this function is executied without any SLS
+targets, a minion will download the :ref:`top file <states-top>` and attempt to
+match the expressions within it. When the minion does match an expression the
+modules listed for it will be downloaded, compiled, and executed.
+
+.. note::
+    This action is referred to as a "highstate".
```
Comment (cachedout) [inline doc/topics/tutorials/states_pt1.rst]: I think we need to expand this documentation to include an explanation stating that `state.apply` is what was previously known as `state.highstate` and has been aliased to make it easier to understand.

**Comment by cachedout** [comment]
I think we just need to add an explanation that we're transitioning to the new language, both in the docs and perhaps in the release notes as well.

**Comment by terminalmage** [comment]
@cachedout Good point, I adjusted the docs per your recommendations in https://github.com/saltstack/salt/pull/32038/commits/74ee8c5

I also moved the highstate usage details above the state.sls usage details in the ``state.apply`` docstring. The thinking here being that people hitting the docs for ``state.apply`` via the docs will more often than not be doing so in the context of a highstate action, so making them scroll down to read docs relevant to this doesn't seem like a good user experience. Finally, I referenced the "highstate" section of the States Tutorial from the ``state.apply`` docstring.

**Pull request merged**